python/housinginsights/sources/base_project.py

```diff
@@ -271,64 +249,112 @@ def create_project_subsidy_csv(self, uid, project_fields_map,
+                    # mar_id_found is the mar_id used if it got to the
+                    # mar_id phase.
+                    nlihc_id, data_id, in_proj_table, found_via, mar_ids_found \
+                        = self._get_nlihc_id_from_db(addresses=addresses,
+                                                     uid=uid)
+
+                    uid_addre = uid + '_addre'  # correct uid for *_addre
+
+                    logger.info('proj_address lookup result: found= {}, '
+                                'via= {}, nlihc_id= {}, '
+                                'unique_data_id origin= {}, current uid= {}, '
+                                'addresses= {}, mar_ids= {}'.format(
+                        in_proj_table, found_via, nlihc_id, data_id, uid_addre,
+                        addresses, mar_ids_found))
+
+                    # add values to the project table output file if no match
+                    # found or match is found and same uid
+                    if not in_proj_table or data_id == uid_addre:
                         data = self._map_data_for_row(nlihc_id=nlihc_id,
                                                       fields=PROJ_FIELDS,
                                                       fields_map=project_fields_map,
                                                       line=source_row)
-                        proj_writer.writerow(data)       
+
+                        # TODO need to finish this method
+                        data = self._geocode_if_needed(data, mar_ids_found)
+
+                        proj_writer.writerow(data)
 
                         # add all to the proj_addre table
-                        # TODO only doing this here means we are assuming that the new source 
+                        # TODO only doing this here means we are assuming that the new source
                         # does not have any additional addresses in its list. Should verify
-                        # that is typical. Otherwise, append_addresses will need to double 
-                        # check that it is not creating duplicate records - which will also 
+                        # that is typical. Otherwise, append_addresses will need to double
+                        # check that it is not creating duplicate records - which will also
                         # be tricky since we will want duplicates from the same source that
-                        # have been loaded into the database previously. 
-                        self._append_addresses(addresses=addresses, nlihc_id = nlihc_id, addre_writer = addre_writer)            
+                        # have been loaded into the database previously.
+
+                        # pass '' if address missing
+                        if not addresses:
+                            addresses = ['']
+                        # TODO: probably not a good idea to add with missing
+                        # TODO: address metadata - we should consider to mar
```

Comment:
Can you explain what you mean by this comment?

Reply:
Specifically for dchousing data we have projects that have mar_id metadata but no valid address - this is a known issue. I was just jotting a quick note that we should consider doing the geocode clean up sooner than later because I had to add code to handle the missing address metadata so it didn't break what I was doing..

Based on our conversations yesterday, the TODO can be removed since we seem to be both on the same page regarding this general issue and discussed briefly about how to best handle geocode clean up process.